mmvb_backend/benchmarking_sessions/api/views.py

```diff
@@ -0,0 +1,124 @@
+from django.shortcuts import get_object_or_404
+from rest_framework import status
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from rest_framework.schemas.openapi import AutoSchema
+from rest_framework.viewsets import ModelViewSet
+
+from benchmarking_sessions.api.serializers import (
+    BenchmarkingSessionResultsSerializer,
+    BenchmarkingSessionSerializer,
+)
+from benchmarking_sessions.models import (
+    BenchmarkingSession,
+    BenchmarkingStepStatus,
+)
+from benchmarking_sessions.tasks import run_benchmark
+
+
+# todo: properly document endpoints
+class BenchmarkingSessionViewSet(ModelViewSet):
+    schema = AutoSchema(tags=["BenchmarkingSessions",])
+    serializer_class = BenchmarkingSessionSerializer
+
+    def get_queryset(self):
+        return BenchmarkingSession.objects.all()
+
+    # todo: generate 404
+    @action(methods=["post"], detail=True, url_path="run")
+    def run_benchmark(self, request, *args, **kwargs):
+        benchmarking_session = get_object_or_404(
```

Comment:
Have you tested this? Since this is a pure django shortcut function I would guess it will raise `Http404` that would return the default error page (so I guess not a json response?): https://docs.djangoproject.com/en/3.0/topics/http/views/#django.http.Http404

I would do it like this:
```
benchmark_id = kwargs["pk"]
try:
    benchmarking_session = BenchmarkingSession.objects.get(id=benchmark_id)
except BenchmarkingSession.DoesNotExist:
    error = f"Benchmark with id {benchmark_id} could not be found"
    return Response({"detail": error}, status=status.HTTP_404_NOT_FOUND
else:
    #  run the task and return response
    pass
```

Reply:
from what i understand django REST framework handles this for us. here's [the doc](https://www.django-rest-framework.org/api-guide/exceptions/).
i've just tested it with CURL and it gave me `{"detail":"Not found."}` so i think it's alright? (if we wanted a more verbose/explicit error message as the one you have in your snippet maybe we should focus on overriding the error handling – as described in the doc – instead of having these manual error handling methods all over the place?)